excellent/functions/functions.go

```diff
@@ -21,107 +21,105 @@ import (
-	"word_slice":        InitialTextFunction(1, 3, WordSlice),
-	"field":             InitialTextFunction(2, 2, Field),
-	"clean":             OneTextFunction(Clean),
-	"left":              TextAndIntegerFunction(Left),
-	"lower":             OneTextFunction(Lower),
-	"right":             TextAndIntegerFunction(Right),
-	"regex_match":       InitialTextFunction(1, 2, RegexMatch),
-	"text_compare":      TwoTextFunction(TextCompare),
-	"repeat":            TextAndIntegerFunction(Repeat),
-	"replace":           ThreeTextFunction(Replace),
-	"upper":             OneTextFunction(Upper),
-	"percent":           OneNumberFunction(Percent),
-	"url_encode":        OneTextFunction(URLEncode),
-
-	// bool functions
-	"and": ArgCountCheck(1, -1, And),
-	"if":  ThreeArgFunction(If),
-	"or":  ArgCountCheck(1, -1, Or),
-
-	// number functions
-	"round":        OneNumberAndOptionalIntegerFunction(Round, 0),
-	"round_up":     OneNumberAndOptionalIntegerFunction(RoundUp, 0),
-	"round_down":   OneNumberAndOptionalIntegerFunction(RoundDown, 0),
-	"max":          ArgCountCheck(1, -1, Max),
-	"min":          ArgCountCheck(1, -1, Min),
-	"mean":         ArgCountCheck(1, -1, Mean),
-	"mod":          TwoNumberFunction(Mod),
-	"rand":         NoArgFunction(Rand),
-	"rand_between": TwoNumberFunction(RandBetween),
-	"abs":          OneNumberFunction(Abs),
-
-	// datetime functions
-	"parse_datetime":      ArgCountCheck(2, 3, ParseDateTime),
-	"datetime_from_epoch": OneNumberFunction(DateTimeFromEpoch),
-	"datetime_diff":       ThreeArgFunction(DateTimeDiff),
-	"datetime_add":        DateTimeAdd,
-	"replace_time":        ArgCountCheck(2, 2, ReplaceTime),
-	"tz":                  OneDateTimeFunction(TZ),
-	"tz_offset":           OneDateTimeFunction(TZOffset),
-	"now":                 NoArgFunction(Now),
-	"epoch":               OneDateTimeFunction(Epoch),
-
-	// date functions
-	"date_from_parts": ThreeIntegerFunction(DateFromParts),
-	"weekday":         OneDateFunction(Weekday),
-	"today":           NoArgFunction(Today),
-
-	// time functions
-	"parse_time":      ArgCountCheck(2, 2, ParseTime),
-	"time_from_parts": ThreeIntegerFunction(TimeFromParts),
-
-	// URN functions
-	"urn_parts": OneTextFunction(URNParts),
-
-	// json functions
-	"json":       OneArgFunction(JSON),
-	"parse_json": OneTextFunction(ParseJSON),
-
-	// formatting functions
-	"format_date":     ArgCountCheck(1, 2, FormatDate),
-	"format_datetime": ArgCountCheck(1, 3, FormatDateTime),
-	"format_time":     ArgCountCheck(1, 2, FormatTime),
-	"format_location": OneTextFunction(FormatLocation),
-	"format_number":   FormatNumber,
-	"format_urn":      OneTextFunction(FormatURN),
-
-	// utility functions
-	"length":     OneArgFunction(Length),
-	"default":    TwoArgFunction(Default),
-	"legacy_add": TwoArgFunction(LegacyAdd),
-	"read_chars": OneTextFunction(ReadChars),
-	"extract":    InitialArrayFunction(1, -1, Extract),
+func init() {
+	std := map[string]types.XFunction{
```

Comment:
apparently i was creating an `initialization loop` so moving this into `init()` avoids that